pkg/util/pool/bucketed_pool.go

```diff
@@ -76,8 +82,14 @@ func (p *BucketedPool[T, E]) Put(s T) {
 	}
 
 	bucketIndex := bits.Len(size - 1)
+	if bucketIndex >= len(p.buckets) {
+		return // Ignore slices larger than the largest bucket
+	}
```

Comment:
Instead of this, should we require that `maxSize` is a power of two in `NewBucketedPool`? I think that would be easier to reason about and less confusing for someone using this type. 

For example, imagine someone created a pool with `maxSize` set to 1000. However, only slices with capacity up to 512 will be pooled, which is likely not what they intended.

Reply:
We should do both. It's possible for somebody to pass in something unexpected.